### PR TITLE
MDOT Address Page- Remove Perm Address Radio Btn If No Temp Address

### PR DESCRIPTION
**Pull request by maharielrosario**
## Description
This PR focuses on conditionally hiding the permanent address radio button until a temporary address is added.

## Screenshots

screenshot-localhost_3001-2020 05 06-16_34_04

screenshot-localhost_3001-2020 05 06-16_41_47

## Acceptance criteria
- [ ] The permanent address radio button should be hidden until a temporary address is added

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/disability-benefits/2346/components/ReviewCardField.jsx [hunk 1]
```diff
@@ -284,8 +283,13 @@ class ReviewCardField extends React.Component {
       'vads-u-margin-top--1',
       'vads-u-width--auto',
     ].join(' ');
-
+    const { data } = this.props;
     const { street, city, country } = this.props.formData;
+    const copyOfTempAddress = data.temporaryAddress;
```
Comment (bradl3yC) [inline src/applications/disability-benefits/2346/components/ReviewCardField.jsx:288]: this isnt necessary as its just a reference to the same place in memory.  You can call ```Object.values(data.temporaryAddress)```

File: src/applications/disability-benefits/2346/components/ReviewCardField.jsx [hunk 2]
```diff
@@ -284,8 +283,13 @@ class ReviewCardField extends React.Component {
...
+    delete copyOfTempAddress['view:livesOnMilitaryBaseInfo'];
```
Comment (bradl3yC) [inline src/applications/disability-benefits/2346/components/ReviewCardField.jsx:289]: you can do this with destructuring

```
const temp = { ['view:livesOnMilitaryBaseInfo]: removed, ...temporaryAddress }
const isTempAddressMissing = Object.values(temporaryAddress).every(
      prop => !prop,
    );
```
['view:livesOnMilitaryBaseInfo'] is being aliased into a variable called removed and everything else in the object is being spread into a variable called temporaryAddress

Comment (mr0sari0) [reply src/applications/disability-benefits/2346/components/ReviewCardField.jsx]: Thanks, just updated.

File: src/applications/disability-benefits/2346/components/ReviewCardField.jsx [hunk 3]
```diff
@@ -284,8 +283,17 @@ class ReviewCardField extends React.Component {
       'vads-u-margin-top--1',
       'vads-u-width--auto',
     ].join(' ');
-
+    const { data } = this.props;
     const { street, city, country } = this.props.formData;
+    /* eslint-disable no-unused-vars */
```
[the remark this answers is not in the document]
Comment (mr0sari0) [reply src/applications/disability-benefits/2346/components/ReviewCardField.jsx:288]: Using destructuring to remove`view:livesonMilitaryBaseInfo` property.

**Review by bradl3yC — APPROVED** [review]

File: src/applications/disability-benefits/2346/components/ReviewCardField.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (bkjohnson) [reply src/applications/disability-benefits/2346/components/ReviewCardField.jsx:288]: What about this instead?

```diff
@@ -285,13 +283,8 @@ class ReviewCardField extends React.Component {
     ].join(' ');
     const { data } = this.props;
     const { street, city, country } = this.props.formData;
-    /* eslint-disable no-unused-vars */
-    // using destructuring to remove view:livesOnMilitaryBaseInfo prop
-    const {
-      'view:livesOnMilitaryBaseInfo': removed,
-      ...temporaryAddress
-    } = data.temporaryAddress;
-    /* eslint-enable no-unused-vars */
+    const { ...temporaryAddress } = data.temporaryAddress;
+    delete temporaryAddress['view:LovesOnMilitaryBaseInfo'];
     const isTempAddressMissing = Object.values(temporaryAddress).every(
       prop => !prop,
     );
```

[the remark this answers is not in the document]
Comment (bkjohnson) [reply src/applications/disability-benefits/2346/components/ReviewCardField.jsx:288]: Actually, I think what you've done is fine.

**Pull request merged**